tests/system/apmserver.py

```diff
@@ -208,13 +209,13 @@ def load_docs_with_template(self, data_path, url, endpoint, expected_events_coun
         assert r.status_code == 202, r.status_code
 
         # make sure template is loaded
-        self.wait_until(
-            lambda: self.log_contains("Finished loading index template"),
-            max_timeout=max_timeout)
-        self.wait_until(lambda: self.es.indices.exists(query_index))
+        # self.wait_until(
+        #    lambda: self.log_contains("Finished loading index template"),
+        #    max_timeout=max_timeout)
+        #self.wait_until(lambda: self.es.indices.exists(query_index))
         # Quick wait to give documents some time to be sent to the index
         # This is not required but speeds up the tests
-        time.sleep(0.1)
+        time.sleep(2)
```

Comment:
Can you explain how this speeds up tests? Now every time a template is loaded this for sure waits 2 seconds.

Reply:
apm-server+libbeat needs 1-2 secs (even with the `flush` change) to index a single event in elasticsearch. If we do not wait, the next call (index refresh) will happen _before_ the indexing happens, thus making the refresh ineffective.

Without an index refresh, you need ~4-5 seconds to query a document and "see" it. This way we put a higher lower bound (2 secs), but it almost guarantees that the refresh happens _after_ the index operation, so the following query (`wait_until`...) returns immediately.

This was made worse by libbeat batching events and taking more time to index them.... some tests were taking 30 seconds just to index and query 3 or 4 documents! 

That said, in my local ~0.5-1 seconds sleep seems to give optimal results, but in Jenkins everything is slower (likely due shared resources) so I think I am going to parametrize this to further speed it up locally.  

I hope this makes sense

Reply:
It sure does, thanks for the details